articles/views.py

```diff
@@ -32,3 +35,124 @@ def delete(self, request: Request, *args: Any, **kwargs: Any) -> Response:
             {"message": "Article deleted successfully"},
             status=status.HTTP_204_NO_CONTENT,
         )
+
+
+class CommentListView(generics.GenericAPIView):
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+    serializer_class = CommentSerializer
+    queryset = Article.objects.all()
+    renderer_classes = (JSONRenderer,)
+
+    def post(self, request: Request, slug: str) -> Response:
+        """
+        Allows authenticated users can add comments on articles
+        Args:
+            slug: this a slug for a particular article
+        Returns:
+            code: The return 201 created for success
+        """
+        self.author = get_object_or_404(Profile, user=request.user)
+        self.article = get_object_or_404(Article, slug=slug)
+        start_index = validate_index(request.data.get("highlight_start"), slug)
+        end_index = validate_index(request.data.get("highlight_end"), slug)
+        if start_index and end_index:
```

Comment:
Why not have this functionality in create method in serializer

Reply:
I did not see the need to since I have the validators functionality.